CMakeLists.txt

```diff
@@ -55,21 +55,31 @@ if(MSVC_USE_STATIC_CRT AND MSVC)
   #
   # 5. We "replace with empty string" (i.e. remove) first, then add, so that '/MT'
   # will be present (and present once) even if '/MD' was not.
-
+  message(STATUS "Configuring Static Runtime Library.")
+  if(${CMAKE_CXX_FLAGS} MATCHES ".*/MD.*")
   string(REGEX REPLACE "/MD" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MT")
+  endif()
```

Comment:
nit: add indentation for the if-blocks for readability

Also, just wondering, what motivated adding these if-guards?

Reply:
Adding the if guards prevents build warnings caused by running the REGEX REPLACE multiple times (the C compiler complains if the same switches are specified multiple times with different values. Without the if guards, you get /MD and /MDd defined, which triggers a compiler command line warning.

I don't like build warnings, they upset me, so I got rid of them.